promdump/promdump.go

```diff
@@ -642,18 +763,81 @@ func main() {
 			// ['export_type="node_export"', 'node_prefix="yb-dev-..."'] => '{export_type="node_export",node_prefix="yb-dev-..."}'
 			ybMetric = fmt.Sprintf("{%v}", strings.Join(labels, ","))
 
-			err = exportMetric(ctx, promApi, ybMetric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, metricName)
+			fileCount, err = exportMetric(ctx, promApi, ybMetric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, metricName)
 
 			if err != nil {
 				log.Printf("exportMetric: export of metric %v failed with error %v; moving to next metric", metricName, err)
 				continue
 			}
+			v.fileCount = fileCount
+			fileCount = 0
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMatricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	// tar the file when Tar feature is enable
+
+	if *enableTar {
+		// Set default filename if not provided
+
+		var filename string
+
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default filename
+			filename = generateDefaultFilename()
+		} else {
+			filename = *tarFilename
+			// Given  filename is properly not given (missing extension)
+			tarExtension := ".tar.gz"
+			bz2Extension := ".bz2"
+
+			fileExtension := strings.HasSuffix(filename, tarExtension) || strings.HasSuffix(filename, bz2Extension)
+
+			if !fileExtension {
+
+				fmt.Println("Please prodive proper filename with extension \n")
+
+				return
+
+			}
```

Comment:
Making this a fatal error seems a bit extreme. There's always the `file` utility if we need to figure out what the customer has done.

This also doesn't prevent the customer from doing things like setting `-tar_compression_algorithm "gzip" -tar_filename lol-fooled-you.tar.bz2`, so I'm not sure there's much value in validating the filename at all.